src/main/java/com/betadevels/onlineshopping/db/hibernate/SubscriptionHibernateDAO.java

```diff
@@ -26,6 +28,15 @@ public SubscriptionHibernateDAO(SessionFactory sessionFactory) {
 		return criteria.list();
 	}
 
+	public Subscription getSubscriptionById(Long subscriptionId)
```

Comment:
The baseHibernateDAO already has the method findById to fetch the record using its primary key, right? I don't think you need this extra method which does the exact same thing in a less efficient manner.